### PR TITLE
Fix fetch tool to process chunked ops while collecting stats

### LINKED ISSUE #8677: fetch-tool fails with "Cannot read property 'contents' of undefined"
Any chance you can help me take a look where it fails?

E:\Git\FluidFramework3\packages\tools\fetch-tool>node bin\fluid-fetch "https://www.office.com/launch/fluid/content?auth=2&action=edit&drive=b!fj-qf9E4cEqEN2yG19SC8FA56ZHdGg5Puse1fuvRM9dWf8H6jmHESKEW0E6-yeH1&item=01L3MDHA3N7VZV7X2JY5G2NOFVOR2V3JJ2&file=https:%2F%2Fmicrosoft-my.sharepoint-df.com%2Fpersonal%2Fjegour_microsoft_com&siteUrl=https:%2F%2Fmicrosoft-my.sharepoint-df.com%2Fpersonal%2Fjegour_microsoft_com" --stat

ERROR: TypeError: Cannot read property 'contents' of undefined
    at processOp (E:\Git\FluidFramework3\packages\tools\fetch-tool\dist\fluidAnalyzeMessages.js:452:61)
    at DataStructureAnalyzer.processOp (E:\Git\FluidFramework3\packages\tools\fetch-tool\dist\fluidAnalyzeMessages.js:186:13)
    at Object.printMessageStats (E:\Git\FluidFramework3\packages\tools\fetch-tool\dist\fluidAnalyzeMessages.js:389:30)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async fluidFetchOneFile (E:\Git\FluidFramework3\packages\tools\fetch-tool\dist\fluidFetch.js:33:5)

This is not FFX file, so can't be opened with Office.com, and not sure what version of Fluid was used to create it, but it was created pretty recently

### PR DESCRIPTION
**Pull request by jatgarg**
Fetch tool was not able to process chunked ops structure. So added the ability to do so.
Fixes: https://github.com/microsoft/FluidFramework/issues/8677

### CODE REVIEW

File: packages/tools/fetch-tool/src/fluidAnalyzeMessages.ts [hunk 1]
```diff
@@ -225,6 +226,7 @@ class DataStructureAnalyzer implements IMessageAnalyzer {
     private readonly dataType = new Map<string, string>();
     private readonly dataTypeStats = new Map<string, [number, number]>();
     private readonly objectStats = new Map<string, [number, number]>();
+    private readonly chunkMap = new Map<string, [string[], number]>();
```
Comment (anthony-murphy) [inline packages/tools/fetch-tool/src/fluidAnalyzeMessages.ts:229]: i think using an object like this is much more readable
```suggestion
    private readonly chunkMap = new Map<string, {chunks:string[], totalSize:number}>();
```

Comment (jatgarg) [reply packages/tools/fetch-tool/src/fluidAnalyzeMessages.ts]: Done

File: packages/tools/fetch-tool/src/fluidAnalyzeMessages.ts [hunk 2]
```diff
@@ -490,6 +496,32 @@ function processOp(
...
+                    runtimeMessage.contents = chunks.join("");
+                    runtimeMessage.type = chunk.originalType;
+                    type = chunk.originalType;
+                    totalMsgSize = value[1];
+                    chunkMap.delete(runtimeMessage.clientId);
+                } else {
+                    return;
+                }
+            }
             default: {
```
Comment (vladsud) [inline packages/tools/fetch-tool/src/fluidAnalyzeMessages.ts:525]: So I think the core of the problem (why errors are not easily diagnosable) is that default case.
Ideally default case should use unreachableCase()
Take a look at isRuntimeMessage(), we should group all those values into some type and leverage it to
- Fail if something comes here outside of this type.
- List all cases here one by one
- Use unreachableCase() for anything else - compiler will detect if that type have additions and will force next developer to add it here.

**Review by vladsud — APPROVED** [review]

File: packages/tools/fetch-tool/src/fluidAnalyzeMessages.ts (hunk 2 above)
Comment (jatgarg) [reply packages/tools/fetch-tool/src/fluidAnalyzeMessages.ts]: Done

**Pull request merged**